swarm-derive/tests/test.rs

```diff
@@ -133,6 +133,23 @@ fn custom_event() {
     }
 }
 
+#[test]
+fn custom_crate() {
+    use libp2p as mylibp2p;
+    #[allow(dead_code)]
+    #[derive(NetworkBehaviour)]
+    #[behaviour(crate = "mylibp2p")]
+    struct Foo {
+        ping: ping::Behaviour,
+        identify: identify::Behaviour,
+    }
+
+    #[allow(dead_code)]
+    fn foo() {
+        require_net_behaviour::<Foo>();
+    }
+}
```

Comment:
This test also already passes with current master (if you remove line 141). 
Correct me if I am wrong, but I believe the current `::libp2p` prefix imports libp2p from the crate level, which means that aliases in a module should not be an issue? @mxinden?

I think it is only a problem if the crate is imported in the `Cargo.toml` with a different name.
`serde` on the other hand states for their `crate` attribute that: 
> This is normally only applicable when invoking re-exported Serde derives from a public macro in a different crate.

I am not against adding this parameter, but could we add a test that fails on current master and would be fixed with this? 🙂

Reply:
As far as I understand, the issue is not clashing symbols but renaming the crate in your manifest.

To test that, we'd have to make a separate test crate with a renamed import.

This make me wonder, should we perhaps just require certain symbols to be present for the macro to work?

That would make https://github.com/libp2p/rust-libp2p/pull/3023 easier to implement.

Or perhaps we could make a seperate "derive" prelude? Like `libp2p::swarm_derive_prelude` that includes all the required symbols. Then we can add a configuration knob here that allows you to change where that prelude is coming from.

Reply:
> This make me wonder, should we perhaps just require certain symbols to be present for the macro to work?
> 
> That would make #3023 easier to implement.
> 
> Or perhaps we could make a seperate "derive" prelude? Like `libp2p::swarm_derive_prelude` that includes all the required symbols. Then we can add a configuration knob here that allows you to change where that prelude is coming from.

Actually, don't worry about it, I already have a solution to that.